Sources/SwiftNavigation/Internal/KeyPath+Sendable.swift

```diff
@@ -0,0 +1,19 @@
+#if compiler(>=6)
+  public typealias _KeyPath<Root, Value> = any KeyPath<Root, Value> & Sendable
+  public typealias _WritableKeyPath<Root, Value> = any WritableKeyPath<Root, Value> & Sendable
```

Comment:
Wonder if we make this explicit for readability:

```suggestion
  public typealias _SendableKeyPath<Root, Value> = any KeyPath<Root, Value> & Sendable
  public typealias _SendableWritableKeyPath<Root, Value> = any WritableKeyPath<Root, Value> & Sendable
```

Reply:
Yep good idea, done! I can do that over in TCA too.